packages/ui/src/components/Routes/Frame.tsx

```diff
@@ -30,44 +30,43 @@ function translationsMap(translatables: FrameQuery['stringTranslations']) {
   );
 }
 
-function useTranslations() {
+export function Frame({ children }: PropsWithChildren) {
   const locale = useLocale();
-  const translations = useOperation(FrameQuery).data?.stringTranslations;
-  return {
-    ...translationsMap(translations?.filter(filterByLocale('en')) || []),
-    ...translationsMap(translations?.filter(filterByLocale(locale)) || []),
-  };
-}
-
-export function Frame(props: PropsWithChildren<{}>) {
-  const locale = useLocale();
-  const translations = useTranslations();
-  const messages = Object.fromEntries(
-    Object.keys(translationSources).map((key) => [
-      key,
-      translations[
-        translationSources[key as keyof typeof translationSources]
-          .defaultMessage
-      ] ||
-        translationSources[key as keyof typeof translationSources]
-          .defaultMessage,
-    ]),
-  );
   return (
-    <IntlProvider locale={locale} messages={messages}>
-      <TranslationsProvider>
-        <Header />
-        <main>
-          {useReducedMotion() ? (
-            <>{props.children}</>
-          ) : (
-            <AnimatePresence mode="wait" initial={false}>
-              {props.children}
-            </AnimatePresence>
-          )}
-        </main>
-        <Footer />
-      </TranslationsProvider>
-    </IntlProvider>
+    <Operation id={FrameQuery}>
+      {(result) => {
+        if (result.state === 'success') {
+          const rawTranslations = result.data.stringTranslations || [];
+          const translations = {
+            ...translationsMap(
```

Comment:
Just FYI. There is a big performance difference between `[...arr1, ...arr2]` and `arr1.concat(arr2)`. Can be essential on big arrays.